src/dynamic_bridge.cpp

```diff
@@ -60,23 +104,36 @@ void update_bridge(
   const std::map<std::string, std::string> & ros2_publishers,
   const std::map<std::string, std::string> & ros2_subscribers,
   std::map<std::string, Bridge1to2HandlesAndMessageTypes> & bridges_1to2,
-  std::map<std::string, Bridge2to1HandlesAndMessageTypes> & bridges_2to1)
+  std::map<std::string, Bridge2to1HandlesAndMessageTypes> & bridges_2to1,
+  bool bridge_all_1to2_topics, bool bridge_all_2to1_topics)
 {
   std::lock_guard<std::mutex> lock(g_bridge_mutex);
 
   // create 1to2 bridges
   for (auto ros1_publisher : ros1_publishers) {
     // identify topics available as ROS 1 publishers as well as ROS 2 subscribers
     auto topic_name = ros1_publisher.first;
+    std::string ros1_type_name = ros1_publisher.second;
+    std::string ros2_type_name;
+
     auto ros2_subscriber = ros2_subscribers.find(topic_name);
     if (ros2_subscriber == ros2_subscribers.end()) {
-      continue;
+      if (bridge_all_1to2_topics) {
+        // update the ROS 2 type name to be the anticipate the bridged type
+        bool mapping_found = ros1_bridge::get_1to2_mapping(ros1_type_name, ros2_type_name);
+        if (!mapping_found) {
+          // printf("No known mapping for ROS 1 type '%s'\n", ros1_type_name.c_str());
+          continue;
+        }
+        // printf("topic name '%s' has ROS 2 publishers\n", topic_name.c_str());
+      } else {
+        continue;
+      }
```

Comment:
The nesting is pretty deep with late `else` blocks and `continue`. I would suggest inverting the logic to `continue` early:

```
if (!bridge_all_1to2_topics) {
  continue;
}
bool mapping_found = ros1_bridge::get_1to2_mapping(ros1_type_name, ros2_type_name);
if (!mapping_found) {
  continue;
}
```